ir/type.cpp

```diff
@@ -771,6 +782,21 @@ VectorType::VectorType(string &&name, unsigned elements, Type &elementTy)
   this->elements = elements;
   defined = true;
   children.resize(elements, &elementTy);
+
+  layout.resize(elements);
+  unsigned elemsz = 0;
+  if (elementTy.isPtrType())
+    elemsz = bits_size_t;
+  else if (elementTy.isIntType() || elementTy.isFloatType())
+    elemsz = elementTy.bits();
+  else {
+    assert("Only pointer / integer / float types are allowed for vector elements"
+           && false);
+  }
+  for (unsigned i = 0; i < elements; ++i)
+    // TODO: elemsz with non-multiple of 8 is not supported.
+    layout[i] = { i * (elemsz / 8), elemsz / 8, false };
```

Comment:
Observed that vector types didn't seem to have paddings: see https://godbolt.org/z/tuPT8R